metricflow/query/query_parser.py

```diff
@@ -643,59 +679,90 @@ def _parse_metric_names(
                     metric_references.extend(list(input_metrics))
         return tuple(metric_references)
 
-    def _parse_linkable_element_names(
+    def _parse_linkable_elements(
         self,
-        qualified_linkable_names: Sequence[str],
         metric_references: Sequence[MetricReference],
+        qualified_linkable_names: Optional[Sequence[str]] = None,
+        linkable_elements: Optional[Sequence[QueryParameter]] = None,
     ) -> QueryTimeLinkableSpecSet:
         """Convert the linkable spec names into the respective specification objects."""
-        qualified_linkable_names = [x.lower() for x in qualified_linkable_names]
+        assert not (qualified_linkable_names and linkable_elements)
+
+        structured_names: List[StructuredLinkableSpecName] = []
+        if qualified_linkable_names:
+            qualified_linkable_names = [x.lower() for x in qualified_linkable_names]
+            structured_names = [StructuredLinkableSpecName.from_name(name) for name in qualified_linkable_names]
+        elif linkable_elements:
+            for linkable_element in linkable_elements:
+                parsed_name = StructuredLinkableSpecName.from_name(linkable_element.name)
+                if parsed_name.time_granularity:
```

Comment:
Interesting. Are we hitting this?

Seems like this validation should be done in the query param objects but I guess we're not there yet.

Reply:
Yeah... This is something I can fix. I'm just learning that there aren't actually any implementations of the `QueryParam` protocol yet, so I can add an implementation class to this PR and put the validation in there.